eclipse_workflow.md

```diff
@@ -0,0 +1,305 @@
+directory reduces clutter.
+
+A typical set of build directories could then look like this
+
+| Directory |  Purpose |
+|:---------|:-------|
+| `bld_master_nompi` | production version for laptop, only compiled from master |
+| `bld_fixes_nompi`  | testing code in branches for small fixes |
+| `bld_fixes_mpi`  | testing code in branches for small fixes with MPI  |
+| `bld_debug_nompi`  | for debugging  |
+
+If you have longer-running branches for major changes, you may want to create one or more `bld_` directories for this branch in addition, so that you can always "hop into" work on that branch without having to recompile much code.
+
+
+## Setting up NEST with Eclipse
+
+### Preparations
+
+You should bootstrap and configure NEST as usual. In this document,
+we will first set up the NEST production build
+`bld_master_nompi`. Handling further configurations will be described
+in a later section.
+
+We thus assume the following directory layout:
+
+    $NEST_ROOT/src                          # source code
+    $NEST_ROOT/bld_master_nompi             # build directory
+    $NEST_ROOT/bld_master_nompi/install     # install directory
+
+You should configure, build and install NEST manually once (note that
+I want to build NEST with gcc 5.1 from Homebrew, therefore the
+`CC=gcc-5 CXX=g++-5` arguments to `configure`; NB: Make sure that you
+have checked out the master branch in the `src` directory):
+
+    cd $NEST_ROOT/bld_master_nompi
+    ../src/configure --prefix=$NEST_ROOT/bld_master_nompi/install CC=gcc-5 CXX=g++-5
+	make -j4
+	make install
+	make installcheck
+
+### Project setup
+
+1. `File > New > Makefile project with existing code`
+1. Choose an arbitrary project name
+1. Browse to the `$NEST_ROOT/src` directory
+1. Keep `C` and `C++` checked
+1. Choose `GNU Autotools Toolchain` for indexer settings.
+1. Click `Finish`
+
+The indexer will scan the code, this may take a while.
+
+To make Eclipse aware of configuration-dependent settings, especially
+include guards such as `HAVE_GSL`, we need to add header files from
+the build directory. To this end, select the project in the project
+browser and choose`Properties` from the context menu. Then
+
+1. go to `C/C++ General > Paths and Symbols`
+1. choose `Includes` tab and there `GNU C`
+1. click `Add`
+1. check off for `Add to all languages`
+1. click `File system ...` and select the
+`$NEST_ROOT/bld_master_nompi/libnestutil` directory
+1. add the `$NEST_ROOT/bld_master_nompi/nest` directory in the same way
+1. rebuild the index when Eclipse suggest it or by choosing `Index >
+   Rebuild` from the context menu on the project.
+
+To enable code formatting with `clang-format` via `CppStyle`, open the Properties window
+for the project and go to `C/C++ General > Formatter`, enable project specific settings, choose `CppStyle` as Code Formatter. `Source > Format` will now format source code according to the `.clang-format` file shipped with NEST.
+
```

Comment:
CppStyle complains that it does not find a proper `.clang-format` file. You have to copy it into the workspace:

    cp $NEST_ROOT/src/.clang-format $HOME/eclipse/workspace/